### PR TITLE
V2.x

### PR DESCRIPTION
**Pull request by Pana**
Update docs

### CODE REVIEW

**Comment by tejasmanohar** [comment]
i'm +1 on most of this. i'll review some more tomorrow since a lot changed. things aren't perfect, but the docs (especially wording of thing) are something that'll evolve over time

**Comment by Pana** [comment]
My english is not good, some sentence may need change, Thanks for point out

**Comment by tejasmanohar** [comment]
@Pana - no worries. thx for contributing :)

**Comment by Pana** [comment]
I think koa's most outstanding feature is no callback, better error handling. By the way is there an better way to call promise then callback ?

**Comment by tejasmanohar** [comment]
>By the way is there an better way to call promise then callback ?
Not sure what you mean here. Code examples? :)

**Comment by Pana** [comment]
I mean is there a better way to express `then callback`

**Comment by tejasmanohar** [comment]
>next is an function, invoke it will run the downstream middleware and get an promise, you can run some code in it's then callback. 

to

>`next` is an function that is invoked to execute the downstream middleware. It returns a Promise

Promise = then-able -> doesn't need to be stated, imo... but if you think it does ->

>`next` is an function that is invoked to execute the downstream middleware. It returns a "then-able" Promise

or

>`next` is an function that is invoked to execute the downstream middleware. It returns a Promise with a with a then function for running code after completion.

But yeah, wouldn't call it a callback since that term generally refers to a function argument that runs at a specific time (usually after the main task).

**Comment by Pana** [comment]
Thx

File: Readme.md [hunk 1]
```diff
@@ -41,12 +39,21 @@ $ npm install koa
  - [Workshop](https://github.com/koajs/workshop) - A workshop to learn the basics of koa, Express' spiritual successor.
  - [Introduction Screencast](http://knowthen.com/episode-3-koajs-quickstart-guide/) - An introduction to installing and getting started with Koa
 
-## Example
+
+## Middleware
+Koa is an middleware framework, it can take 3 different kind function as middleware:
```
Comment (jmealo) [inline Readme.md]: This should read "3 different kinds"

**Comment by jonathanong** [comment]
oh shit this is awesome guys. i'll try to take a look more tomorrow!

**Comment by jonathanong** [comment]
shall we just merge this now and continuously improve?

**Comment by tejasmanohar** [comment]
@jonathanong yea, way better than what's there now.

ps: wtf github? i don't think i pushed that :P
wtf github

**Pull request closed without merging**